cpp/compilation-db.rst

```diff
@@ -0,0 +1,28 @@
+
+
+#######################################
+Creating a compilation database
+#######################################
+
+``scons`` does not natively support the creation of a compilation database, instead the
+``bear`` tool can be used (`bear repository  <https://github.com/rizsotto/Bear>`_) 
+which is available on a number of Linux distributions (`bear packages <https://repology.org/project/bear/packages>`_) as well as
+from conda-forge.
+
+====================================
+Install Bear from conda-forge
+====================================
+To install ``bear`` into a lsst conda stack
+.. code-block:: bash
+
+    conda install -c conda-forge
+
+=====================================================
+Use Bear and Scons to create a compilation database
+=====================================================
+
+.. code-block:: bash
+
+   bear -- scons
+
+This will create ``compile_commands.json`` containing a list of all compilation commands that can be used in various editors.
```

Comment:
Should we add instructions on how to configure editors to use it (if any configuration is needed)?

Reply:
Created a link to `VScode` extensions as a `scons` hack is mentioned here to create a compilations database. This sections in `VScode` should be cleaned up and link to the `bear` section. `VScode` is using the CDB if present and appropriate extensions are installed.